consensus/src/chained_bft/persistent_liveness_storage.rs

```diff
@@ -45,6 +45,10 @@ pub trait PersistentLivenessStorage<T>: Send + Sync {
     /// to jump to this round
     fn save_highest_timeout_cert(&self, highest_timeout_cert: TimeoutCertificate) -> Result<()>;
 
+    /// Retrieve a validator change proof for SafetyRules so it can instantiate its
+    /// ValidatorVerifier.
+    fn retrieve_validator_change_proof(&self, version: u64) -> Result<ValidatorChangeProof>;
```

Comment:
this is previously done by state_computer.get_epoch_proof

Reply:
That is different as it expects epochs, we could try it out and see what happens, but let's assume we start safety rules on a new validator -- the system is at epoch 10000 with a modern waypoint... safety rules will start at epoch 0 and not be able to make sense of what to do next. And consensus would still need to somehow figure out how to convert the version to an LI.

It isn't entirely clear to me why we're asking state computer to perform this versus the persistent store?

Reply:
it doesn't matter which perform this, just want to unify it

Reply:
Perhaps we could move the cleanup to another PR. I think we need to look at the consensus APIs and consider a broader simplification / unification. It seems really awkward for consensus to be querying executor when it could inquire directly into storage. I don't know what we gain by the indirection.